libraries/AC_PID/AC_PID_2D.cpp

```diff
@@ -133,16 +133,17 @@ void AC_PID_2D::update_i(const Vector2f &limit)
     _pid_info_y.limit = false;
 
     Vector2f delta_integrator = (_error * _ki) * _dt;
-    if (!is_zero(limit.length_squared())) {
-        // zero delta_vel if it will increase the velocity error
+    float integrator_length = _integrator.length();
+    _integrator += delta_integrator;
+    if (!is_zero(integrator_length)) {
```

Comment:
I don't think we need this check. looks to me that`_integrator.limit_length` correctly deals with the 0 case.

Removing the check means we can remove the `integrator_length` float too.

It also means that we correctly set the pid_info limit flags even if there is 0 I term. Its a edge case but this could happen on the first loop after a I reset if you have a large I gain and small I max.

Reply:
This is done but I don't see how we can remove integrator_length.